cloudsql-postgresql-plugin/src/e2e-test/java/io/cdap/plugin/cloudsqlpostgresql/BQValidation.java

```diff
@@ -0,0 +1,268 @@
+                        Date dateTarget = java.sql.Date.valueOf(
+                                bigQueryData.get(jsonObjectIdx).get(columnName).getAsString());
+                        Assert.assertEquals("Different values found for column : %s", dateSource, dateTarget);
+                        break;
+
+                    case Types.DOUBLE:
+                        Double sourceMoney = rsSource.getDouble(currentColumnCount);
+                        String targetMoneyStr = bigQueryData.get(jsonObjectIdx).get(columnName).getAsString();
+                        Double targetMoney;
+                        // Remove non-numeric characters from the targetMoneyStr
+                        targetMoneyStr = targetMoneyStr.replaceAll("[^0-9.]", "");
+                        targetMoney = new Double(targetMoneyStr);
+                        Assert.assertTrue(String.format("Different values found for column: %s", columnName),
+                                sourceMoney.compareTo(targetMoney) == 0);
+                        break;
+
+                    case Types.VARCHAR:
+                    case Types.CHAR:
+                    case Types.SQLXML:
+                    case Types.OTHER:
+                    default:
+                        String sourceValue = rsSource.getString(currentColumnCount);
+                        JsonElement jsonElement = bigQueryData.get(jsonObjectIdx).get(columnName);
+                        String targetValue = (jsonElement != null && !jsonElement.isJsonNull()) ? jsonElement.getAsString() : null;
+                        Assert.assertEquals(
+                                String.format("Different %s values found for column : %s", columnTypeName, columnName),
+                                String.valueOf(sourceValue), String.valueOf(targetValue));
+                }
+                currentColumnCount++;
+            }
+            jsonObjectIdx++;
+        }
+        Assert.assertFalse("Number of rows in Source table is greater than the number of rows in Target table",
+                rsSource.next());
+        return true;
+    }
+
+    private static boolean getBooleanValidation(ResultSet rsSource, String bqDateString, String columnName,
+                                                String columnTypeName) throws SQLException {
+        switch (columnTypeName) {
+            case "bit":
+                byte source = rsSource.getByte(columnName);
+                boolean sourceAsBoolean = source != 0;
+                return String.valueOf(sourceAsBoolean).equals(String.valueOf(bqDateString));
+            case "bool":
+                boolean sourceValue = rsSource.getBoolean(columnName);
+                return String.valueOf(sourceValue).equals(String.valueOf(bqDateString));
+            default:
+                return false;
+        }
+    }
+
+    private static boolean getTimeValidation(ResultSet rsSource, String bqDateString, String columnName, String
```

Comment:
Is this method getting used somewhere now ?

Reply:
Yes we are using this !

Reply:
you are using this code ```Timestamp sourceTS = rsSource.getTimestamp(columnName);
                        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'");
                        Date parsedDate = dateFormat.parse(bigQueryData.get(jsonObjectIdx).get(columnName).getAsString());
                        Timestamp targetTs = new Timestamp(parsedDate.getTime());
                        Assert.assertEquals(sourceTS, targetTs);```
                        
  To comapre time stamp data. Then what is this method used for ?

Reply:
We are using getTimeValidation for Time with and without zone data not for Timestamp